code/modules/reagents/reactions/other.dm

```diff
@@ -275,3 +225,92 @@
 	name = "laughter"
 	results = list(/datum/reagent/consumable/laughter = 5)
 	required_reagents = list(/datum/reagent/consumable/sugar = 1, /datum/reagent/consumable/drink/banana = 1)
+	
+	
+//Explosives and pyrotechnics
+
+
+/datum/chemical_reaction/emp_pulse
+	name = "EMP Pulse"
+	required_reagents = list(/datum/reagent/uranium = 1, /datum/reagent/iron = 1) // Yes, laugh, it's the best recipe I could think of that makes a little bit of sense
+
+/datum/chemical_reaction/emp_pulse/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	// 100 created volume = 4 heavy range & 7 light range. A few tiles smaller than traitor EMP grandes.
+	// 200 created volume = 8 heavy range & 14 light range. 4 tiles larger than traitor EMP grenades.
+	empulse(location, round(created_volume / 24), round(created_volume / 14), 1)
+	holder.clear_reagents()
+
+
+/datum/chemical_reaction/flash_powder
+	name = "Flash powder"
+	required_reagents = list(/datum/reagent/aluminum = 1, /datum/reagent/potassium = 1, /datum/reagent/sulfur = 1)
+
+/datum/chemical_reaction/flash_powder/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+	s.set_up(2, 1, location)
+	s.start()
+	for(var/mob/living/carbon/M in viewers(WORLD_VIEW, location))
+		switch(get_dist(M, location))
+			if(0 to 3)
+				if(M.flash_act())
+					M.Paralyze(30 SECONDS)
+
+			if(4 to 5)
+				if(M.flash_act())
+					M.Stun(10 SECONDS)
+
+
+/datum/chemical_reaction/napalm
+	name = "Napalm"
+	required_reagents = list(/datum/reagent/aluminum = 1, /datum/reagent/toxin/phoron = 2, /datum/reagent/toxin/acid = 1 )
+
+/datum/chemical_reaction/napalm/on_reaction(datum/reagents/holder, created_volume, radius)
+	radius = round(sqrt(created_volume * 0.15)) //allows a nice, healthy 3-tile fire if using 2 120u beakers fully filled up.
+	if(radius < 0) radius = 0
```

Comment:
```suggestion
	if(radius < 0)
		radius = 0
```

Reply:
but what is flame_radius(0) doing?

Reply:
I don't know why I left that there. It's not like it can be *less* than 0.

Reply:
Ok, so what is flame radius (0) actually doing?